### ISSUE TITLE
Add [Obsolete] attribute for System.Data.SqlClient in .NET Core and above

### ISSUE THREAD

**Issue by alexandrehtrb** [dotnet/runtime#81495]
Hello,

[This article](https://devblogs.microsoft.com/dotnet/introducing-the-new-microsoftdatasqlclient/) suggests that System.Data.SqlClient is deprecated, in .NET Core and above, in favor of Microsoft.Data.SqlClient.
Should there be an `[Obsolete]` attribute in System.Data.SqlClient classes, to encourage people to use Microsoft.Data.SqlClient?

**Comment by JRahnama** [comment]
@alexandrehtrb S.D.SqlClient is not deprecated. Per documentation

> What does this mean for System.Data.SqlClient?
It means the development focus has changed. We have no intention of dropping support for System.Data.SqlClient any time soon. It will remain as-is and we will fix important bugs and security issues as they arise. If you have a typical application that doesn’t use any of the newest SQL features, then you will still be well served by a stable and reliable System.Data.SqlClient for many years.

> However, Microsoft.Data.SqlClient will be the only place we will be implementing new features going forward. We would encourage you to evaluate your needs and options and choose the right time for you to migrate your application or library from System.Data.SqlClient to Microsoft.Data.SqlClient.

In other words, we encourage people to move to M.D.SqlClient which the active ADO.NET library and gets updated on regular basis, whereas S.D.SqlClient is in servicing mode and does not get any update but for security reason.
